### PR TITLE
Fix four recent issues

### PR DESCRIPTION
**Pull request by awarzon**
*Issue #, if available:*
#15 solved with https://github.com/aws-quickstart/quickstart-trek10-serverless-enterprise-cicd/commit/07b61c1beed6dc95b21cba424599a03c0bbb4cf3

#16 solved with https://github.com/aws-quickstart/quickstart-trek10-serverless-enterprise-cicd/commit/dabcee4d271f6176226caa649df527aaf23dee63

#17 and #20 solved with https://github.com/aws-quickstart/quickstart-trek10-serverless-enterprise-cicd/commit/786202ddb08fcc318b4d678d1bc9bbfc9ebc52cd

*Description of changes:*

#15 handling long stack names in dynamic branch stacks

#16 better error handling in StackMakerLambda

#17 & #20: Updating deprecated nodejs runtime in sample app

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: templates/full-stack.template.yaml [hunk 1]
```diff
@@ -63,7 +63,7 @@ Parameters:
   BuildImageName:
     Description: Docker image for application build
     Type: String
-    Default: aws/codebuild/nodejs:10.1.0
+    Default: lambci/lambda:build-nodejs12.x
```
Comment (santiagocardenas) [inline templates/full-stack.template.yaml]:
```suggestion
    Default: aws/codebuild/amazonlinux2-x86_64-standard:2.0	
```

Then specify the `nodejs: 12` runtime in the buildspec.yml as shown in https://docs.aws.amazon.com/codebuild/latest/userguide/build-spec-ref.html#runtime-versions-buildspec-file

**Comment by avattathil** [comment]
@santiagocardenas please approve when this PR ready to merge

File: templates/full-stack.template.yaml (hunk 1 above)
Comment (awarzon) [reply templates/full-stack.template.yaml]: Thanks, added in https://github.com/aws-quickstart/quickstart-trek10-serverless-enterprise-cicd/pull/22/commits/8d96e818f8535be87380d91afb3e613d927fc3db

**Comment by santiagocardenas** [comment]
Thank you @awarzon !

@avattathil LGTM please.

**Pull request merged**